src/python/pants/build_graph/target.py

```diff
@@ -632,6 +638,40 @@ def traversable_dependency_specs(self):
     """
     return []
 
+  @classmethod
+  def compute_dependency_specs(cls, kwargs=None, payload=None):
+    """Given either pre-Target.__init__() kwargs or a post-Target.__init__() payload, compute the
+    full set of dependency specs in the same vein as the prior `traversable_dependency_specs`.
+
+    N.B. This is a temporary bridge to span the gap between v2 "Fields" products vs v1 `BuildGraph`
+    `Target` object representations. See:
+
+      https://github.com/pantsbuild/pants/issues/3560
+      https://github.com/pantsbuild/pants/issues/3561
+
+    :param dict kwargs: The pre-Target.__init__() kwargs dict.
+    :param Payload payload: The post-Target.__init__() Payload object.
+    :yields: Spec strings representing dependencies of this target.
```

Comment:
Is there any reason not to return `Address` objects here? Ditto on the Subsystem/injectables. Pre-parsing them once at creation time would make more sense, but I can't remember if there was some reason to use strings.